### ISSUE TITLE
How are the regridding weights generated

### ISSUE THREAD

**Issue by TomNicholas** [IAOCEA/xarray-healpy#17]
Hi @keewis - I was having a quick look at this package and was wondering how you generate the regridding weights? Is it just with ESMF? And then the [regridder](https://github.com/IAOCEA/xarray-healpy/blob/main/xarray_healpy/regridder.py) uses those but applies them using sparse matrices?

**Comment by keewis** [comment]
no, this makes use of a custom [`numba`-based](https://github.com/IAOCEA/xarray-healpy/blob/main/xarray_healpy/interpolation/bilinear.py) algorithm.

In theory, the advantage of the custom implementation is that we could create a `dask`-aware version (if we can figure out how to perform the neighbours search with `dask`). On the other hand this also means that all the there's all sorts of issues / edge cases that don't work properly, yet.

**Comment by TomNicholas** [comment]
Thanks - interesting.

This is custom in the sense that it is specific to the HealPix grid? Or is it actually general?

**Comment by keewis** [comment]
it is supposed to be general, but that totally depends on the algorithm used for the neighbours search. Right now, that is a ball tree with an appropriate metric – for example euclidean or haversine. But bilinear interpolation requires convex quadrilaterals (or just not self-intersecting? not sure), which I can't really guarantee with the current implementation.

So when I said "custom" I was referring to the fact that it is reimplementing bilinear interpolation (so that was probably not the right word).
